src/OneExplorer/OneStorage.ts

```diff
@@ -24,12 +25,157 @@ import {Logger} from '../Utils/Logger';
 import {ConfigObj} from './ConfigObject';
 import {Node, NodeType} from './OneExplorer';
 
-interface StringListMap {
-  [key: string]: string[];
+export {
+  BaseModelToCfgMap as _unit_test_BaseModelToCfgMap,
+  CfgToCfgObjMap as _unit_test_CfgToCfgObjMap,
+};
+
+class CfgToCfgObjMap {
+  private _map: Map<string, ConfigObj>;
+
+  constructor() {
+    this._map = new Map<string, ConfigObj>();
+  }
+
+  public init(cfgList: string[]) {
+    cfgList.forEach(cfg => {
+      const cfgObj = ConfigObj.createConfigObj(vscode.Uri.file(cfg));
+      if (cfgObj) {
+        this._map.set(cfg, cfgObj);
+      }
+    });
+  }
+
+  get size() {
+    return this._map.size;
+  }
+
+  public get(path: string) {
+    return this._map.get(path);
+  }
+
+  public reset(type: NodeType, path: string) {
+    switch (type) {
+      case NodeType.config:
+        this._map.delete(path);
+        break;
+      case NodeType.baseModel:
+      case NodeType.product:
+      case NodeType.directory:
+      default:
+        assert.isOk(false, `Cannot reach here`);
+        break;
+    }
+  }
+
+  /**
+   * Update the data when cfg file's path is changed or the content is changed.
+   * @param type NodeType
+   * @param path config path to update
+   * @param newpath (optional) if exists, change the file path
```

Comment:
I cannot fully understand this method to review. 😓 When will this method be called without the `newpath`? AFAIU it will reload the `ConfigObj` with the exactly same path before.

Reply:
This map is designed to handle the events defined in OneExplorer.

> When will this method be called without the newpath?

- When a config file is changed and `onChangedFile` event is triggered -> update the CfgObj (newpath === undefined) with the same old key
- When a config file is renamed and `onDidRenameFile` event is triggered -> update the CfgObj (newpath !== undefined) with the new key